### ISSUE TITLE
quickMove doesn't fully complete on Tb 128

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#480]
This is a new regression in Thunderbird 128: When using quickMove to move emails, the emails are moved as expected but the label (and the staged list with the number) do remain visible in the UI:

![image](https://github.com/user-attachments/assets/c81c3a46-24ad-406c-83b5-e63dd009355c)

The only way to get rid of them at the moment is to use the right-click menu and the command "Cancel quickMove"

![image](https://github.com/user-attachments/assets/94636bd7-6059-43a9-85b6-edf63eea058c)

**Comment by RealRaven2000** [comment]
There seems to be a silent crash in `MailServices.copy.copyMessages()` - this may be possibly caused by changes in `nsIMsgCopyServiceListener`  (which modifies some spellings)

see [Bug 1887047](https://bugzilla.mozilla.org/show_bug.cgi?id=1887047) 

the changes landed in Thunderbird 126

**Comment by RealRaven2000** [comment]
Here is a quick fix:

[QuickFolders-mx-6.5.2pre40.zip](https://github.com/user-attachments/files/16247864/QuickFolders-mx-6.5.2pre40.zip)

-----

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

----

***detail of the fix***

some of the spellings had to be changed for the `nsIMsgCopyServiceListener` service. This version is also backwards compatible with 115.

main patch:

```js
          // [issue 480] - [Bug 1887047]
          const modernCopyService =  (util.versionGreaterOrEqual(util.ApplicationVersion, "126"));

          let p = await new Promise((resolve, reject) => {
            // AUTF8String getMessageId()
            const param5 =
              modernCopyService 
              ?
              {
                onStartCopy() {},
                onProgress(progress, progressMax) {},
                setMessageKey(key) {},
                getMessageId(messageId) {
                  return null;
                },
                onStopCopy(status) {
                  if (status == Cr.NS_OK) {
                    debugger;
                    resolve(); // returns to await
                  } else {
                    debugger;
                    reject(status); // this throws !
                  }
                }
              }
              :
              {
                OnStartCopy() {},
                OnProgress(progress, progressMax) {},
                SetMessageKey(key) {},
                GetMessageId(messageId) {},
                OnStopCopy(status) {
                  if (status == Cr.NS_OK) {
                    debugger;
                    resolve(); // returns to await
                  } else {
                    debugger;
                    reject(status); // this throws !
                  }
                }
              }            


              MailServices.copy.copyMessages(
                sourceFolder,
                msgHeaders,
                targetFolder,
                isMove && sourceFolder.canDeleteMessages,
                param5,
                mw,  // msgWindow
                true   // allowUndo
              );
            });


```

**Comment by RealRaven2000** [comment]
Fixed in 6.6 - Published 21/07/2024
